wallet-new/integration/TransactionSpecs.hs

```diff
@@ -47,6 +48,60 @@ transactionSpecs wRef wc = do
 
             map txId resp `shouldContain` [txId txn]
 
+        it "posted deposits appear in index for asset locked wallets" $ do
+            genesis <- genesisWallet wc
+            (fromAcct, _) <- firstAccountAndId wc genesis
+
+            wallet <- genesisAssetLockedWallet wc
+            (toAcct, toAddr) <- firstAccountAndId wc wallet
+
+            let payment = Payment
+                    { pmtSource =  PaymentSource
+                        { psWalletId = walId genesis
+                        , psAccountIndex = accIndex fromAcct
+                        }
+                    , pmtDestinations = pure PaymentDistribution
+                        { pdAddress = addrId toAddr
+                        , pdAmount = tenthOf (accAmount fromAcct)
+                        }
+                    , pmtGroupingPolicy = Nothing
+                    , pmtSpendingPassword = Nothing
+                    }
+                tenthOf (V1 c) = V1 (Core.mkCoin (Core.getCoin c `div` 10))
+
+            etxn <- postTransaction wc payment
+
+            txn <- fmap wrData etxn `shouldPrism` _Right
+
+            eresp <- getTransactionIndex wc (Just (walId wallet)) (Just (accIndex toAcct)) Nothing
+            resp <- fmap wrData eresp `shouldPrism` _Right
+
+            map txId resp `shouldContain` [txId txn]
+
+        it "asset-locked transactions fail to submit" $ do
+            genesis <- genesisAssetLockedWallet wc
+            (fromAcct, _) <- firstAccountAndId wc genesis
+
+            wallet <- sampleWallet wRef wc
+            (_toAcct, toAddr) <- firstAccountAndId wc wallet
+
+            let payment = Payment
+                    { pmtSource =  PaymentSource
+                        { psWalletId = walId genesis
+                        , psAccountIndex = accIndex fromAcct
+                        }
+                    , pmtDestinations = pure PaymentDistribution
+                        { pdAddress = addrId toAddr
+                        , pdAmount = tenthOf (accAmount fromAcct)
+                        }
+                    , pmtGroupingPolicy = Nothing
+                    , pmtSpendingPassword = Nothing
+                    }
+                tenthOf (V1 c) = V1 (Core.mkCoin (Core.getCoin c `div` 10))
+
+            etxn <- postTransaction wc payment
```

Comment:
I think the problem here is that it does not test what we want to test. It checks that we cannot add the transaction into the mempool of the local node, and this fails and we then do not even try to send the transaction to another node. So this is a purely local operation. It does not check that if we do actually send such a transaction to another node (especially a core node) that the transaction is rejected.

That kind of test is much harder to set up of course. It'd likely have to construct the transaction and then submit it directly to a peer node (e.g. like how the auxx tool does). Given the current structure of the code it is perhaps enough to rely on the fact that the local validation for adding to the local mempool is the same validation used by core nodes when they receive a transaction. Sadly this will not necessarily always be the case, e.g. the new wallet may simply post transactions directly and not send them via the mempool of the local node. Also, in the new wallet, adding a pending transaction will not synchronously send it to a peer node, that'll be done asynchronously. So there will be no failure result to check here.